src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * A class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    protected String filePath;
+
+    /**
+     * Constructs a new Storage class initialized with the path of the target file.
+     *
+     * @param filePath the path of the file to load and store.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves the list of tasks to the target file.
+     *
+     * @param taskList the list of tasks.
+     * @throws DukeException thrown if the file or folder does not exist or other IOException happens.
+     */
+    public void store(List<Task> taskList) throws DukeException {
+        File outFile;
+        Writer out;
```

Comment:
hmm 'out' seems a little ambiguous here maybe can rename for better readability?